provider/datagen/src/cldr/transform/calendar/japanese.rs

```diff
@@ -48,6 +48,17 @@ impl TryFrom<&dyn CldrPaths> for JapaneseErasProvider {
     }
 }
 
+impl TryFrom<&crate::DatagenOptions<'_>> for JapaneseErasProvider {
+    type Error = eyre::ErrReport;
+    fn try_from(options: &crate::DatagenOptions) -> eyre::Result<Self> {
+        JapaneseErasProvider::try_new(
+            options
+                .cldr_paths
+                .ok_or_else(|| eyre::eyre!("JapaneseErasProvider requires cldr_paths"))?,
```

Comment:
though: This error message is not very useful for a user. `DatagenOptions` should provide this (by exposing an `eyre::Result<impl CldrPaths>`) so that it can set something like `"...requires either --cldr_root or ... to be set"`).

Reply:
If I make a centralized function like `pub fn get_cldr_paths(&self) -> eyre::Result<&dyn cldr::CldrPaths>`, then the error message becomes less useful, because I don't have the name of the transformer. Can you be more specific with your suggestion, maybe with a little code sample of what you are envisioning?

Reply:
That is what I mean. I think you're thinking of this as an internal error message, while I think this is user-facing, because `datagen.rs` in the future should not set the fields for which the user doesn't specify arguments.

Do you agree that as a user-facing message, `JapaneseEraProvider requires cldr_paths` is not good because it's not actionable and exposes internal implementation details instead of the CLI API?

The best of both worlds would be to provider an error message like
`list/and@1 requires CLDR data. Either --cldr-tag or --cldr-root or --input-from-testdata must be specified.`

We can do this by changing the signature to `pub fn get_cldr_paths(&self, key: ResourceKey) -> eyre::Result<&dyn cldr::CldrPaths>` and generating the error message with the correct key if required.

Reply:
In that case I don't see the point of making the fields `Option`s right now, because they are currently always populated, but let's just go and merge this for now.

Reply:
> In that case I don't see the point of making the fields `Option`s right now, because they are currently always populated, but let's just go and merge this for now.

I see. If this is where the contention originated, I'm happy to undo the last couple of commits to go back to non-optional fields in DatagenOptions.

Reply:
If you agree that this is not the end state let's just merge